adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpAdapterSaslAuthenticatorFactory.java

```diff
@@ -200,7 +201,16 @@ private void verifyPlain(final byte[] saslResponse, final Handler<AsyncResult<De
                     completer.handle(Future.failedFuture(new CredentialException(
                             "username does not comply with expected pattern [<authId>@<tenantId>]")));
                 } else {
-                    getUsernamePasswordAuthProvider().authenticate(credentials, completer);
+                    getTenantObject(credentials.getTenantId()).map(tenant -> {
+                        if (tenant.isAdapterEnabled(Constants.PROTOCOL_ADAPTER_TYPE_AMQP)) {
+                            getUsernamePasswordAuthProvider().authenticate(credentials, completer);
```

Comment:
we should be able to do both in parallel:
* getting the Tenant
* getting the Credentials

by using a CompositeFuture. However, we can postpone that optimization til later ...

Reply:
ok, I will take a look